### PR TITLE
Allow Django 3

### PR DESCRIPTION
**Pull request by bgrant**
Allow Django < 4 in requirements.txt

The tests pass, but there are warnings (now fixed).

```
WARNINGS:
django_declarative_apis.OauthConsumer: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tests.TestModel: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
```

More warnings when running `$ python -Wa manage.py test`:

```
/Users/robert.grant/.pyenv/versions/django-declarative-apis/lib/python3.7/site-packages/kombu/utils/compat.py:82: DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
  for ep in importlib_metadata.entry_points().get(namespace, [])
Creating test database for alias 'default'...
/Users/robert.grant/dev/django-declarative-apis/tests/urls.py:17: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
  url(r"^simple", resource_adapter(get=views.SimpleEndpointDefinition)),
/Users/robert.grant/dev/django-declarative-apis/tests/urls.py:18: RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().
  url(r"^dict", resource_adapter(get=views.DictEndpointDefinition)),
System check identified some issues:
```

### CODE REVIEW

**Comment by bgrant** [comment]
https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys

> Maintaining the historical behavior, the default value for DEFAULT_AUTO_FIELD is AutoField. Starting with 3.2 new projects are generated with DEFAULT_AUTO_FIELD set to BigAutoField.

**Comment by bgrant** [comment]
These changes as-is seem to be enough for upgrading Django to 3.2, but they break the `test-all` target (the part where we recurse into the example directory and install packages there).  I'm not sure the "test" target for the example directory were working before.

**Comment by rohansroy** [comment]
> RemovedInDjango40Warning: django.conf.urls.url() is deprecated in favor of django.urls.re_path().

Looks like our upgrade to Django 4 will be a little more involved. I'm surprised they're making this change, seems somewhat unpythonic.

File: example/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ../../django-declarative-apis
-Django~=2.2.0
+Django >=2.2.0, <4
```
Comment (rohansroy) [inline example/requirements.txt:2]: We can probably make this `>=3.2` once we upgrade Django in VE ya?

Comment (bgrant) [reply example/requirements.txt:2]: Yeah- I guess 2.2 won't be supported by Django anymore, so there's no reason for us to support it

Comment (rohansroy) [reply example/requirements.txt:2]: do we want `>=` then?

**Review by rohansroy — DISMISSED** [review]

**Comment by bgrant** [comment]
`test-all` also runs the example tests.  For those to work, you have to run `make example-install` first, which will install `sslserver` from `example`'s `requirements.txt`.

File: example/requirements.txt (hunk 1 above)
Comment (bgrant) [reply example/requirements.txt:2]: Good point.  Let me ask the team.

Comment (bgrant) [reply example/requirements.txt]: (I dropped 2.2)

File: setup.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 HERE = path.dirname(__file__)
 
 with open(path.join(HERE, "requirements.txt")) as requirements_file:
-    requirements = requirements_file.read().split()
+    requirements = requirements_file.read().strip().split("\n")
```
Comment (rohansroy) [inline setup.py:16]: why is this necessary now?

Comment (bgrant) [reply setup.py:16]: When you install DDA with `pip install`, it installs the list of dependencies indicated by `install_requires`, which this is used for further down.  The `example` package installs DDA this way.  It was erroring out when the dependency was spelled:

```
django >=2.2, <4
```

because this parsing didn't handle those spaces correctly.  It's not strictly necessary now that I changed the dependency to

```
django~=3.2
```

but I still think it's a bug in our setup.py, even if it's not triggered now.

**Comment by bgrant** [comment]
Closes #83

**Review by rohansroy — APPROVED** [review]
